### PR TITLE
Show (truncated) configmap data in overview

### LINKED ISSUE #199: Overview: Configmap display should show data
Currently, the display for ConfigMaps in `kubernetes-overview` simply shows the namespace and creation date, like so:
```
  metadata-grpc-configmap                            2    69d
    Namespace:  namespace
    Created:    2021-06-18T17:59:07Z
```

We could expand this to display the ConfigMap's data contents.

```
  metadata-grpc-configmap                            2    69d
    Namespace:  namespace
    Created:    2021-06-18T17:59:07Z
    Data: (2)
      HOST: localhost
      PORT: 8080
```

Care might be needed here for particularly large data entries, e.g. ConfigMaps that themselves contain entire text config files in a data entry. In such cases, we might display a truncated version.

### PR DESCRIPTION
**Pull request by Walheimat**
Closes #199.

Shows configmap data in a new section in the overview.

Both keys and values are truncated. If the value spans multiple lines, only the first line is shown (and maybe truncated). To show this truncation, a backward slash (dimmed) is shown.

Adds a new case to the configmap test response; the expected case is updated accordingly.

*Note* that for me the tests don't work, so I had to run the updated test using `cask exec ert-runner ./test/kubernetes-configmaps-test.el` (which succeeds):

```sh
➜  kubernetes-el git:(feature/cm-data-in-overview) make test
cask clean-elc
cask exec ert-runner
Debugger entered--Lisp error: (error "Test ‘kubernetes-contexts-test--empty-state’ redef...")
  error("Test `%s' redefined" kubernetes-contexts-test--empty-state)
  ert-set-test(kubernetes-contexts-test--empty-state #s(ert-test :name kubernetes-contexts-test--empty-state :documentation nil :body (closure (t) nil (let ((temp-buffer (generate-new-buffer " *temp*" t))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (save-excursion ...) (let* ... ...) (search-forward-regexp "Context:[[:space:]]+") (let* ... ...)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))))) :most-recent-result nil :expected-result-type :passed :tags nil))
  load-with-code-conversion("/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." nil t)
  load("/home/walheimat/emacs/kubernetes-el/test/kubernete..." nil :nomessage)
  ert-runner--load("/home/walheimat/emacs/kubernetes-el/test/kubernete...")
  mapc(ert-runner--load ("/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..."))
  -each(("/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete..." "/home/walheimat/emacs/kubernetes-el/test/kubernete...") ert-runner--load)
  (let ((test-files (ert-runner--test-files tests)) (test-helper (f-expand "test-helper.el" ert-runner-test-path))) (-each ert-runner-load-files #'ert-runner--load) (if (f-exists\? test-helper) (ert-runner--load test-helper)) (-each test-files #'ert-runner--load) (if ert-runner-verbose (ert-runner/run-tests-batch-and-exit ert-runner-selector) (let ((shut-up-sink (generate-new-buffer " *shutup*")) (inhibit-message t)) (let* ((--cl-shut-up-current-output-- #'(lambda nil ...))) (if shut-up-ignore (progn (ert-runner/run-tests-batch-and-exit ert-runner-selector)) (unwind-protect (let* (... ... ... ... ... ...) (unwind-protect ... ... ... ...)) (and (buffer-name shut-up-sink) (kill-buffer shut-up-sink))))))))
  ert-runner/run()
  apply(ert-runner/run nil)
  commander--handle-command(nil)
  commander-parse(nil)
  (if commander-parsing-done nil (commander-parse (or commander-args (cdr command-line-args-left))))
  load-with-code-conversion("/home/walheimat/emacs/kubernetes-el/.cask/29.0/elp..." "/home/walheimat/emacs/kubernetes-el/.cask/29.0/elp..." nil t)
  command-line-1(("-scriptload" "/home/walheimat/emacs/kubernetes-el/.cask/29.0/elp..."))
  command-line()
  normal-top-level()

make: *** [Makefile:64: test] Error 255
```

### CODE REVIEW

**Comment by Walheimat** [comment]
Looks like I missed something, will update later.

File: kubernetes-utils.el [hunk 1]
```diff
@@ -73,6 +73,15 @@ initialized."
       (concat (substring s 0 (1- threshold)) "...")
     s))
 
+(defun kubernetes-utils-ellipsize-multiline (s threshold)
```
Comment (jinnovation) [inline kubernetes-utils.el:76]: Could we extend the existing `kubernetes-utils-ellipsize` to handle multi-line?

**Review by jinnovation — CHANGES_REQUESTED** [review]
Thanks for looking into this!

Let's also add an entry to the CHANGELOG to reflect these new additions.

**Comment by jinnovation** [comment]
> so the error comes from a redefinition. 

Do you have a sense of how that specific test (`kubernetes-contexts-test--empty-state`) is getting redefined? Curious what changes in Emacs 29 to cause this.

> Should I fix this in another PR so it works for 29?

If the fix can't be bundled into this PR, then follow-up PR is fine. Since we don't provide full support for Emacs 29, as long as CI passes we're good.

**Comment by Walheimat** [comment]
If this change is okay with you, I'd rebase into a single commit.

I'll look at the test redefinition in another PR.

**Comment by jinnovation** [comment]
> If this change is okay with you, I'd rebase into a single commit.

No need; I've restricted PRs to squash-and-merge only, so no need to worry about excessive commits "polluting" repo history.

> I'll look at the test redefinition in another PR.

Sounds good; thanks.

File: kubernetes-utils.el [hunk 2]
```diff
@@ -69,9 +69,19 @@ initialized."
          value)))))
 
 (defun kubernetes-utils-ellipsize (s threshold)
-  (if (> (length s) threshold)
-      (concat (substring s 0 (1- threshold)) "...")
-    s))
+  "Ellipsize a string.
```
Comment (jinnovation) [inline kubernetes-utils.el:72]:
```suggestion
  "Ellipsize a string S that's longer than THRESHOLD.
```

**Review by jinnovation — APPROVED** [review]

**Pull request merged**